### PR TITLE
Add back accidentally removed API

### LINKED ISSUE #150: API break in #135 (1.3.0)
In spite of https://github.com/apple/swift-log/pull/135#issuecomment-647621205, I think the was API actually broken in #135:

This:

```swift
@inlinable
public func debug(_ message: @autoclosure () -> Logger.Message,
                  metadata: @autoclosure () -> Logger.Metadata? = nil,
                  file: String = #file, function: String = #function, line: UInt = #line) {
    ...
}
```

...is not the same as:

```swift
@inlinable
public func debug(_ message: @autoclosure () -> Logger.Message,
                  metadata: @autoclosure () -> Logger.Metadata? = nil,
                  source: @autoclosure () -> String? = nil,
                  file: String = #file, function: String = #function, line: UInt = #line) {
    ...
}
```

We need to retain the old function signatures in addition to the new ones and just forward the implementation from the old to the new.

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

- When `source:` was added in #135, the API was accidentally broken
  despite source compatability.

Modifications:

- Add back source-less API calls which defer to the decls with source.
- Note we can't deprecate the calls we're adding back:
  `log.info("some message")` refers to the implementation we're adding
  back rather and would therefore require users to explicitly specify a source
  to suppress the deprecation warning message.

Result:

- API is no longer broken.
- Resolves #150

### CODE REVIEW

**Comment by glbrntt** [comment]
API breakages between 1.2.0 and 1.3.0:

```
❯ xcrun --toolchain org.swift.53202008101a ./scripts/check_no_api_breakages.sh . 1.3.0 1.2.0
Cloning into '/tmp/.check-api_zfKAaT/repo'...
done.
Checking public API breakages from 1.2.0 to 1.3.0
[4/4] Merging module Logging
[4/4] Merging module Logging
Checking Logging.json... ERROR
==============================
ERROR: public API change in Logging.json
==============================

/* Generic Signature Changes */

/* RawRepresentable Changes */

/* Removed Decls */
Func Logger.log(level:_:metadata:file:function:line:) has been removed
Func MultiplexLogHandler.log(level:message:metadata:file:function:line:) has been removed
Func StreamLogHandler.log(level:message:metadata:file:function:line:) has been removed

/* Moved Decls */

/* Renamed Decls */
Func Logger.critical(_:metadata:file:function:line:) has been renamed to Func critical(_:metadata:source:file:function:line:)
Func Logger.debug(_:metadata:file:function:line:) has been renamed to Func debug(_:metadata:source:file:function:line:)
Func Logger.error(_:metadata:file:function:line:) has been renamed to Func error(_:metadata:source:file:function:line:)
Func Logger.info(_:metadata:file:function:line:) has been renamed to Func info(_:metadata:source:file:function:line:)
Func Logger.notice(_:metadata:file:function:line:) has been renamed to Func notice(_:metadata:source:file:function:line:)
Func Logger.trace(_:metadata:file:function:line:) has been renamed to Func trace(_:metadata:source:file:function:line:)
Func Logger.warning(_:metadata:file:function:line:) has been renamed to Func warning(_:metadata:source:file:function:line:)

/* Type Changes */
Func Logger.critical(_:metadata:file:function:line:) has parameter 2 type change from Swift.String to () -> Swift.String?
Func Logger.critical(_:metadata:file:function:line:) has parameter 4 type change from Swift.UInt to Swift.String
Func Logger.debug(_:metadata:file:function:line:) has parameter 2 type change from Swift.String to () -> Swift.String?
Func Logger.debug(_:metadata:file:function:line:) has parameter 4 type change from Swift.UInt to Swift.String
Func Logger.error(_:metadata:file:function:line:) has parameter 2 type change from Swift.String to () -> Swift.String?
Func Logger.error(_:metadata:file:function:line:) has parameter 4 type change from Swift.UInt to Swift.String
Func Logger.info(_:metadata:file:function:line:) has parameter 2 type change from Swift.String to () -> Swift.String?
Func Logger.info(_:metadata:file:function:line:) has parameter 4 type change from Swift.UInt to Swift.String
Func Logger.notice(_:metadata:file:function:line:) has parameter 2 type change from Swift.String to () -> Swift.String?
Func Logger.notice(_:metadata:file:function:line:) has parameter 4 type change from Swift.UInt to Swift.String
Func Logger.trace(_:metadata:file:function:line:) has parameter 2 type change from Swift.String to () -> Swift.String?
Func Logger.trace(_:metadata:file:function:line:) has parameter 4 type change from Swift.UInt to Swift.String
Func Logger.warning(_:metadata:file:function:line:) has parameter 2 type change from Swift.String to () -> Swift.String?
Func Logger.warning(_:metadata:file:function:line:) has parameter 4 type change from Swift.UInt to Swift.String

/* Decl Attribute changes */

/* Fixed-layout Type Changes */

/* Protocol Conformance Change */

/* Protocol Requirement Change */
Func LogHandler.log(level:message:metadata:source:file:function:line:) has been added as a protocol requirement

/* Class Inheritance Change */
done
```


API breakages between 1.2.0 and 1.3.0:

**Comment by glbrntt** [comment]
API breakages between 1.2.0 and this PR:

```
❯ xcrun --toolchain org.swift.53202008101a ./scripts/check_no_api_breakages.sh . HEAD 1.2.0
Cloning into '/tmp/.check-api_4E9ITe/repo'...
done.
Checking public API breakages from 1.2.0 to HEAD
[4/4] Merging module Logging
[4/4] Merging module Logging
Checking Logging.json... ERROR
==============================
ERROR: public API change in Logging.json
==============================

/* Generic Signature Changes */

/* RawRepresentable Changes */

/* Removed Decls */
Func MultiplexLogHandler.log(level:message:metadata:file:function:line:) has been removed
Func StreamLogHandler.log(level:message:metadata:file:function:line:) has been removed

/* Moved Decls */

/* Renamed Decls */

/* Type Changes */

/* Decl Attribute changes */

/* Fixed-layout Type Changes */

/* Protocol Conformance Change */

/* Protocol Requirement Change */
Func LogHandler.log(level:message:metadata:source:file:function:line:) has been added as a protocol requirement

/* Class Inheritance Change */
done
```

**Comment by glbrntt** [comment]
I think these two are okay because of the default implementation on the `LogHandler`

```
Func MultiplexLogHandler.log(level:message:metadata:file:function:line:) has been removed
Func StreamLogHandler.log(level:message:metadata:file:function:line:) has been removed
```

And this one is also fine for the same reason: it has a default implementation:

```
Func LogHandler.log(level:message:metadata:source:file:function:line:) has been added as a protocol requirement
```

File: Tests/LoggingTests/LoggingTest.swift [hunk 1]
```diff
@@ -711,14 +775,14 @@ class LoggingTest: XCTestCase {
...
-        logger.error(Dummy())
+        logger.error(error: Dummy())
 
         logging.history.assertExist(level: .error, message: "errorDescription")
     }
 }
 
 extension Logger {
-    public func error(_ error: Error,
+    public func error(error: Error,
```
Comment (glbrntt) [inline Tests/LoggingTests/LoggingTest.swift:785]: fyi: I had to rename this so `let error = logger.error(_:metadata:file:function:line:)` wasn't ambiguous

**Comment by ktoso** [comment]
(Sorry for being slow to get to this, will do soon)

**Comment by glbrntt** [comment]
> (Sorry for being slow to get to this, will do soon -- want to make sure it's really correct this time)

No worries; I'm in no rush for this!

**Review by ktoso — APPROVED** [review]
Oh my, I was pretty bad to follow up here but LGTM.

Going to release with this today

**Pull request merged**